crates/telio-wg/src/wg.rs

```diff
@@ -677,9 +688,23 @@ impl State {
                     )
                     .await?;
                 }
+
+                if self.should_emit_downgrade_events
+                    && link_detection_update_result.should_notify
```

Comment:
Both `should_emit_downgrade_events` and `should_notify` look like they mean the same thing, are they both needed?

Reply:
self.should_emit_downgrade_events - the config says we use link detection for downgrade
link_detection_update_result.should_notify - the link state of the peer has been changed

Reply:
Sorry, I don't understand :/

Reply:
So, the first one, self.should_emit_downgrade_events, indicates that libtelio is configured to use link detection for downgrade. The feature flag use_for_downgrade is set.
The second one is the result of link_detection.update() which indicates that the link_state has changed at this update call.